python/activator/activator.py

```diff
@@ -257,73 +324,95 @@ def keda_start():
+
+                        redis_streams_message_id, fan_out_visit_decoded = _decode_redis_streams_message(
+                            fan_out_message)
+
+                        # Calculate time to receive message based on timestamp in Redis Stream message
+                        fan_out_to_prompt_time = _calculate_time_since_fan_out_message_delivered(
+                            redis_streams_message_id)
+                        _log.debug("Seconds since fan out message delivered %r", fan_out_to_prompt_time)
+
+                        expected_visit = FannedOutVisit.from_dict(fan_out_visit_decoded)
+                        _log.debug("Unpacked message as %r.", expected_visit)
+
+                        # Ack the redis stream message and close redis stream client
+                        # TODO Consider moving xack after process visit completes for catch up processing.
+                        redis_client.xack(redis_stream_name,
+                                          redis_stream_consumer_group,
+                                          redis_streams_message_id)
+                        redis_client.close()
+                # TODO Review Redis Errors and determine what should be retriable.
+                except redis.exceptions.RedisError as e:
+                    _log.critical("Redis Streams error; aborßting.")
+                    _log.exception(e)
+                    redis_client.close()
+                    sys.exit(1)
+
+                with instances_processing_gauge.track_in_progress():
+                    try:
+                        # Process fan out visit
+                        process_visit(expected_visit)
+                    except Exception as e:
+                        _log.critical("Process visit failed; aborting.")
+                        _log.exception(e)
+                    finally:
+                        _log.info(
+                            "Processing completed for %s.  Starting next fan out event consumer poll",
+                            socket.gethostname())
+
+                # Reset timer for fan out message polling and start redis client for next poll
+                fan_out_listen_start_time = time.time()
                 try:
-                    # Process fan out visit
-                    process_visit(deserialized_fan_out_visit)
+                    redis_client = _get_redis_streams_client()
+                    redis_client.ping()
+                    _log.info("Redis Streams client setup for continued polling")
                 except Exception as e:
-                    _log.critical("Process visit failed; aborting.")
+                    _log.critical("Redis Streams client unexpected error in continued polling; aborting")
                     _log.exception(e)
-                finally:
-                    _log.info("Processing completed for %s", socket.gethostname())
-                    break
+                    redis_client.close()
```

Comment:
I think this is fine so long as `Redis.close` is idempotent -- if `_get_redis_streams_client()` raises, then `redis_client` exists but points to the old object.